modules/indexing/src/test/java/org/apache/ignite/internal/processors/database/WalDisabledDuringIndexRecreateTest.java

```diff
@@ -0,0 +1,404 @@
+import static org.apache.ignite.configuration.DataStorageConfiguration.DFLT_WAL_PATH;
+import static org.apache.ignite.configuration.DataStorageConfiguration.UNLIMITED_WAL_ARCHIVE;
+import static org.apache.ignite.internal.pagemem.PageIdAllocator.INDEX_PARTITION;
+import static org.apache.ignite.internal.processors.cache.GridCacheUtils.cacheGroupId;
+import static org.apache.ignite.internal.processors.cache.GridCacheUtils.cacheId;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.CACHE_DIR_PREFIX;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.CACHE_GRP_DIR_PREFIX;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.DFLT_STORE_DIR;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.INDEX_FILE_NAME;
+
+/** */
+@RunWith(Parameterized.class)
+public class WalDisabledDuringIndexRecreateTest extends GridCommonAbstractTest {
+    /** Batches count. */
+    public static final int ENTRIES_CNT = 1_000;
+
+    /** */
+    public static final String GRP_NAME = "my-group";
+
+    /** */
+    public static final int GRP_CACHES_CNT = 3;
+
+    /** */
+    public static final long UNKNOWN_PAGE_ID = -1;
+
+    /** */
+    private ListeningTestLogger testLog;
+
+    /** */
+    @Parameterized.Parameter()
+    public boolean cacheGrps;
+
+    /** */
+    @Parameterized.Parameters(name = "cacheGroups={0}")
+    public static Iterable<Object> data() {
+        return Arrays.asList(true, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setGridLogger(testLog)
+            .setConsistentId(igniteInstanceName)
+            .setClusterStateOnStart(INACTIVE)
+            .setFailureHandler(new StopNodeFailureHandler())
+            .setDataStorageConfiguration(new DataStorageConfiguration()
+                .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                    .setPersistenceEnabled(true))
+                .setMaxWalArchiveSize(UNLIMITED_WAL_ARCHIVE));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** */
+    @Test
+    public void testDisabled() throws Exception {
+        FileWALPointer walStartPtr = createDataAndDeleteIndexBin();
+
+        testLog = new ListeningTestLogger(log);
+
+        awaitRebuild();
+
+        assertEquals(
+            0,
+            countWalGrpRecords(wp -> ((FileWALPointer)wp).compareTo(walStartPtr) > 0, CU.cacheGroupId(cacheName(), cacheGroupName()))
+        );
```

Comment:
probably we need additional check that index is in operable\enabled state after rebuid ? and further new kv updates will touch index at all